### PR TITLE
kcov: migrate to Conan v2

### PR DESCRIPTION
**Pull request by valgur**
TODO

- [x] Merge #17272.

### CODE REVIEW

**Comment by valgur** [comment]
Closing temporarily to avoid unnecessary load on the CI. Will reopen when I'm actively working on the PR again.

**Review by jwillikers — APPROVED** [review]

File: recipes/kcov/all/conandata.yml [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
     sha256: "6b1c11b066d57426d61375a31c3816f1fcd2610b447050c86d9920e22d5200b3"
   "38":
     url: "https://github.com/SimonKagstrom/kcov/archive/38.tar.gz"
     sha256: "b37af60d81a9b1e3b140f9473bdcb7975af12040feb24cc666f9bb2bb0be68b4"
 patches:
-  "40": []
+  "42":
+    - patch_file: "patches/0003-fix-missing-include.patch"
+      patch_description: "Add a missing includes on macOS"
+      patch_type: "backport"
```
Comment (uilianries) [inline recipes/kcov/all/conandata.yml:15]:
```suggestion
      patch_type: "backport"
      patch_source: "https://github.com/SimonKagstrom/kcov/commit/4efe5e4a4ac4e7b6d99fc869d826aa74ff716ebf"
```

File: recipes/kcov/all/test_package/conanfile.py [hunk 2]
```diff
@@ -1,9 +1,17 @@
...
+    def requirements(self):
+        self.requires(self.tested_reference_str)
+
+    def layout(self):
+        cmake_layout(self)
 
     def test(self):
-        if not tools.cross_building(self.settings):
-            self.run("kcov --version", run_environment=True)
+        self.run("kcov --version", env="conanrun")
```
Comment (uilianries) [inline recipes/kcov/all/test_package/conanfile.py:17]:
```suggestion
        if can_run(self):
            self.run("kcov --version", env="conanrun")
```

File: recipes/kcov/all/test_package/conanfile.py [hunk 3]
```diff
@@ -1,9 +1,17 @@
-from conans import ConanFile, tools
+from conan import ConanFile
+from conan.tools.cmake import cmake_layout
```
Comment (uilianries) [inline recipes/kcov/all/test_package/conanfile.py:2]:
```suggestion
from conan.tools.cmake import cmake_layout
from conan.tools.build import can_run
```

**Review by uilianries — APPROVED** [review]
LGTM. About the adjusted variables, it could be a PR to the upstream, changing to use CMake targets instead :)

**Review by perseoGI — APPROVED** [review]
LGTM

**Review by RubenRBS — APPROVED** [review]
Thanks!

**Pull request merged**